File: app/src/main/java/org/syxc/zhihudaily/api/Callback.java
package org.syxc.zhihudaily.api;

/**
 * Created by syxc on 15/12/15.
 */
public interface Callback<T> {

  /**
   * Success http response
   */
  void onSuccess(T t);

  void onCompleted();

  /**
   * Fail http response
   */
  void onError(String error);
}

File: app/src/main/java/org/syxc/zhihudaily/api/Callback.kt
package org.syxc.zhihudaily.api

/**
 * Callback
 * Created by syxc on 15/12/15.
 */
interface Callback<T> {

  /**
   * Success http response
   */
  fun onSuccess(t: T)

  fun onCompleted()

  /**
   * Fail http response
   */
  fun onError(error: String?)
}
